pkg/scheduler/preemption/preemption.go

```diff
@@ -96,19 +96,30 @@ func (p *Preemptor) GetTargets(wl workload.Info, assignment flavorassigner.Assig
 	if len(sameQueueCandidates) == len(candidates) {
 		// There is no possible preemption of workloads from other queues,
 		// so we'll try borrowing.
-		targets = minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, true)
-	} else {
-		// There is a risk of preemption of workloads from the other queue in the
-		// cohort, proceeding without borrowing.
-		targets = minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, false)
-		if len(targets) == 0 {
-			// Another attempt. This time only candidates from the same queue, but
-			// with borrowing. The previous attempt didn't try borrowing and had broader
-			// scope of preemption.
-			targets = minimalPreemptions(&wl, assignment, snapshot, resPerFlv, sameQueueCandidates, true)
-		}
+		return minimalPreemptions(&wl, assignment, snapshot, resPerFlv, candidates, true, nil)
 	}
 
+	// There is a risk of preemption of workloads from the other queue in the
+	// cohort. We proceed with borrowing only if the dedicated policy
+	// (borrowWithinCohort) is enabled. This ensures the preempted workloads
+	// have lower priority, and so they will not preempt the preemptor when
+	// requeued.
+	var allowBorrowingBelowPriority *int32
+	borrowWithinCohort := cq.Preemption.BorrowWithinCohort
+	allowBorrowing := borrowWithinCohort != nil && borrowWithinCohort.Policy != kueue.BorrowWithinCohortPolicyNever
+	if allowBorrowing {
+		allowBorrowingBelowPriority = ptr.To(priority.Priority(wl.Obj))
+		if borrowWithinCohort.MaxPriorityThreshold != nil && *borrowWithinCohort.MaxPriorityThreshold < *allowBorrowingBelowPriority {
+			allowBorrowingBelowPriority = ptr.To(*borrowWithinCohort.MaxPriorityThreshold + 1)
+		}
```

Comment:
I think it makes more sense to do `return minimalPreemptions` here. It makes it more obvious that there is only one call to `minimalPreemptions`.

Reply:
Good point, thank you for the comment, I think it reads much better now.